pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,351 @@
+		return c.updateStatusToUnknown(plant, message, conditionAPIServerAvailable, conditionEveryNodeReady)
+	}
+
+	plantClusterClient, discoveryClient, err := c.initializePlantClients(plant, key, kubeconfig)
+	if err != nil {
+		message := fmt.Sprintf("Could not initialize Plant clients: %+v", err)
+		return c.updateStatusToUnknown(plant, message, conditionAPIServerAvailable, conditionEveryNodeReady)
+	}
+
+	healthChecker := c.initializeHealthChecker(plantClusterClient, discoveryClient)
+
+	// Trigger health check
+	conditionAPIServerAvailable, conditionEveryNodeReady = c.healthChecks(ctx, healthChecker, logger, conditionAPIServerAvailable, conditionEveryNodeReady)
+
+	cloudInfo, err := FetchCloudInfo(ctx, plantClusterClient, discoveryClient, logger)
+	if err != nil {
+		return err
+	}
+
+	return c.updateStatus(plant, cloudInfo, conditionAPIServerAvailable, conditionEveryNodeReady)
+}
+
+func (c *defaultPlantControl) updateStatusToUnknown(plant *gardencorev1alpha1.Plant, message string, conditionAPIServerAvailable, conditionEveryNodeReady gardencorev1alpha1.Condition) error {
+	conditionAPIServerAvailable = helper.UpdatedCondition(conditionAPIServerAvailable, corev1.ConditionFalse, "APIServerDown", message)
+	conditionEveryNodeReady = helper.UpdatedCondition(conditionEveryNodeReady, corev1.ConditionFalse, "Nodes not reachable", message)
+	return c.updateStatus(plant, &StatusCloudInfo{}, conditionAPIServerAvailable, conditionEveryNodeReady)
+}
+
+func (c *defaultPlantControl) updateStatus(plant *gardencorev1alpha1.Plant, cloudInfo *StatusCloudInfo, conditions ...gardencorev1alpha1.Condition) error {
+	_, err := kutil.TryUpdatePlantStatusWithEqualFunc(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(p *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			if p.Status.ClusterInfo == nil {
+				p.Status.ClusterInfo = &gardencorev1alpha1.ClusterInfo{}
+			}
+			p.Status.ClusterInfo.Cloud.Type = cloudInfo.CloudType
+			p.Status.ClusterInfo.Cloud.Region = cloudInfo.Region
+			p.Status.ClusterInfo.Kubernetes.Version = cloudInfo.K8sVersion
+			p.Status.Conditions = conditions
+			return p, nil
+		}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+			return equality.Semantic.DeepEqual(cur, updated)
+		})
+	return err
+}
+
+func (c *defaultPlantControl) delete(ctx context.Context, plant *gardencorev1alpha1.Plant, logger logrus.FieldLogger) error {
+	if _, err := kutil.TryUpdatePlantWithEqualFunc(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta, func(c *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+		finalizers := sets.NewString(c.Finalizers...)
+		finalizers.Delete(FinalizerName)
+		c.Finalizers = finalizers.UnsortedList()
+		return c, nil
+	}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+		return !sets.NewString(cur.Finalizers...).Has(FinalizerName)
+	}); err != nil {
+		return err
+	}
+
+	// Remove finalizer from referenced secret
```

Comment:
We should first remove the finalizer from the secret before removing it from the plant.